src/autoscaler.py

```diff
@@ -3,20 +3,32 @@
 import logging
 from ops.pebble import ExecError
 from pathlib import Path
-from typing import Dict
+from typing import TypedDict
 import yaml
 
 from errors import JujuEnvironmentError
 
 logger = logging.getLogger(__name__)
 CONTROLLER = "scaler-controller"
+CLOUD_CONFIG_FILE = Path("/", "config", "cloud-config.yaml")
+
+
+ControllerData = TypedDict(
+    "ControllerData",
+    {"api-endpoints": list[str], "uuid": str, "ca-cert": str},
+)
+Secrets = TypedDict("Secrets", {"user": str, "password": str})
+ModelData = TypedDict("ModelData", {"models": dict, "current-model": str})
+CloudConfig = TypedDict(
+    "CloudConfig", {"Username": str, "Password": str, "Endpoint": list[str], "Cacert": str}
```

Comment:
I changed the yaml identifiers for the cloud  config to be more yamly as requested. The yaml identifiers should now be:
```
user
password
endpoints
ca-cert
```